### PR TITLE
ENG-357 add auto generated graphql client

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
add auto-generated graphql client

## Test Plan
| command                         | passed |
|---------------------------------|--------|
| init                            | yes    |
| build                           | yes    |
| deploy                          | yes    |
| diff                            |    yes    |
| create                          |    yes    |
| watch                           |   yes     |
| wait                            |    yes    |
| info                            |     yes  |
| apply                           |   yes     |
| validate                        |      yes  |
| topsort                         |    yes    |
| bounce                          |   yes     |
| destroy                         |  yes      |
| preflights                      |     yes   |
| login                           |     yes   |
| import                          |    yes    |
| repair                          |      yes  |
| serve                           |      yes  |
| shell                           |   yes     |
| repos                           |     yes   |
| push terraform                  | yes    |
| push helm                       |    yes    |
| push recipe                     |     yes   |
| push resourcedefinition         |        |
| push integration                |        |
| push artifact                   |        |
| push dashboard                  |        |
| push crd                        |    yes    |
| api list installations          |   yes     |
| api list charts                 |    yes    |
| api list terraform              |    yes    |
| api list versons                |    yes    |
| api list chartinstallations     |    yes    |
| api list terraforminstallations |    yes    |
| api list artifacts              |    yes    |
| config amend                    |        |
| config read                     |        |
| config import                   |        |
| workspace kube-init             |   yes    |
| workspace helm                  |    yes   |
| workspace helm-diff             |    yes    |
| workspace helm-deps             |    yes    |
| workspace terraform-diff        |    yes    |
| workspace crds                  |    yes   |
| profile use                     |  yes      |
| profile save                    |      yes  |
| profile show                    |     yes   |
| profile list                    |    yes    |
| output                          |   yes    |
| logs                            |    yes    |
| bundle                          |   yes     |
| ops terminate                   |    yes    |
| ops cluster                     |    yes    |
| utils image-bump                |        |
| template                        |    yes    |
| build-context                   |        |
| changed                         |     yes   |
| from-grafana                    |        |

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/api/charts.go [hunk 1]
```diff
@@ -128,30 +65,33 @@ func (client *Client) GetPackageInstallations(repoId string) (charts []*ChartIns
...
 
-	tfs = make([]*TerraformInstallation, len(resp.TerraformInstallations.Edges))
-	for i, edge := range resp.TerraformInstallations.Edges {
-		tfs[i] = edge.Node
+	tfs = make([]*TerraformInstallation, 0)
+	for _, edge := range resp.TerraformInstallations.Edges {
+		tfInstall := &TerraformInstallation{
+			Id: *edge.Node.ID,
+			Terraform: &Terraform{
+				Id:             utils.ConvertStringPointer(edge.Node.Terraform.ID),
```
[the remark this answers is not in the document]
Comment (zreigz) [reply pkg/api/charts.go:84]: good point, util methods will be used in other repos

**Comment by zreigz** [comment]
@michaeljguarino @DavidSpek @floreks @maciaszczykm  I have tested all API calls besides: 
push resourcedefinition
push integration 
push artifact 
push dashboard

PTAL

**Comment by michaeljguarino** [comment]
> @michaeljguarino @DavidSpek @floreks @maciaszczykm I have tested all API calls besides: push resourcedefinition push integration push artifact push dashboard
> 
> PTAL

The only of those we'd need to test should be push artifact, which I don't know if we really use anymore either (the only repo it was used on is in the Pluralfile for `pluralsh/plural-cli` itself which I believe was deprecated)

**Comment by zreigz** [comment]
@michaeljguarino so please let me know if I need to get rid of some of the commands. I can do this in this PR or we can do this later

**Comment by michaeljguarino** [comment]
Yeah let's remove all except for the artifacts one.  We'll also want to remove the targets in `pkg/pluralfile` that reference them as well

**Review by DavidSpek — APPROVED** [review]
Tested all the most used commands, including a fresh deploy on AWS using the GitHub integration and all looks good. Awesome work!

**Pull request merged**